edx_toggles/toggles/internal/waffle/flag.py

```diff
@@ -0,0 +1,179 @@
+            "%sFlag '%s' accessed without a request",
+            self.log_prefix,
+            self.name,
+        )
+        value = _is_flag_active_for_everyone(self.name)
+        set_custom_attribute("warn_flag_no_request_return_value", value)
+        return value
+
+    def set_monitor_value(self, value):
+        """
+        Send waffle flag value to monitoring. We keep this method such that it can be called by child classes (such as
+        edx-platform's waffle_utils.CourseWaffleFlag). Child classes are expected to call this method if they are
+        overriding `is_enabled`, and has cases where it is not calling `super().is_enabled().
+        """
+        _set_waffle_flag_attribute(self.name, value)
+
+
+class NonNamespacedWaffleFlag(WaffleFlag):
+    """
+    Same as the WaffleFlag class, but does not require that the instance name be namespaced. This class is useful for
+    migrating existing Flag objects; new instances should always be namespaced.
+    """
+
+    @classmethod
+    def validate_name(cls, name):
+        pass
+
+
+def _is_flag_active_for_everyone(flag_name):
+    """
+    Returns True if the waffle flag is configured as active for Everyone,
+    False otherwise.
+    """
+    # Import is placed here to avoid model import at project startup.
+    # pylint: disable=import-outside-toplevel
+    from waffle.models import Flag
+
+    try:
+        waffle_flag = Flag.objects.get(name=flag_name)
+        return waffle_flag.everyone is True
+    except Flag.DoesNotExist:
+        return False
+
+
+def _set_waffle_flag_attribute(name, value):
+    """
+    For any flag name in settings.WAFFLE_FLAG_CUSTOM_ATTRIBUTES, add name/value
+    to cached values and set custom attribute if the value changed.
+
+    Important: Remember to configure ``WAFFLE_FLAG_CUSTOM_ATTRIBUTES`` for
+    LMS, Studio and Workers in order to see waffle flag usage in all
+    edx-platform environments.
+
+    .. setting_name: WAFFLE_FLAG_CUSTOM_ATTRIBUTES
+    .. setting_default: False
+    .. setting_description: A set of waffle flags to track with custom attributes having
+      values of (True, False, or Both). The name of the custom attribute will have the prefix ``flag_`` and the suffix
+      will match the name of the flag. The value of the custom attribute could be False, True, or Both.
+
+      The value Both would mean that the flag had both a True and False value at different times during the
+      transaction. This is most likely due to happen in WaffleFlag child classes, such as edx-platform's
+      waffle_utils.CourseWaffleFlag.
+
+      An example NewRelic query to see the values of a flag in different environments, if your waffle flag was named
+      ``my.waffle.flag`` might look like::
+
+        SELECT count(*) FROM Transaction
+        WHERE flag_my.waffle.flag IS NOT NULL
+        FACET appName, flag_my.waffle.flag
+
+    .. setting_warning: This will work if it is a list, but it might be less performant.
```

Comment:
I was originally confused by this.  Maybe the description needs an example of setting the setting?  If it were a dict, the values would just be ignored, right?

Reply:
> Maybe the description needs an example of setting the setting?

I'll make the warning more precise.

> If it were a dict, the values would just be ignored, right?

Yes, absolutely.